### DISCUSSION TITLE
Problem with UNO style in MainPage.xaml

### DISCUSSION THREAD

**Discussion by ward-horsfall** [unoplatform/uno#18914] [Q&A] [answered]
Hi,

What is wrong with my XAML code in MainPage.xaml

TextBox_Source looks different from TextBox_Destination and TextBox_Game_Name.

They should look the same. I want all three to look like TextBox_Destination.

What is wrong with UniformTextBoxStyle?

Thanks

Ward

![Screenshot 2024-11-26 003407](https://github.com/user-attachments/assets/fd03c523-e6df-43a5-82a7-92842fec299a)

`

Code:

<Page x:Class="Mame_Copy_UNO.MainPage"
      xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
      xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
      xmlns:local="using:Mame_Copy_UNO"
      xmlns:utu="using:Uno.Toolkit.UI"
      xmlns:um="using:Uno.Material"
      Background="{ThemeResource BackgroundBrush}">

    
    <Page.Resources>
        <Style TargetType="TextBox" x:Key="UniformTextBoxStyle">
            <Setter Property="Width" Value="500"/>
            <Setter Property="Height" Value="Auto"/>
            <Setter Property="Margin" Value="10,0,0,0"/>
        </Style>
    </Page.Resources>
    <ScrollViewer>
    <Grid utu:SafeArea.Insets="VisibleBounds">
      <StackPanel
        HorizontalAlignment="Center"
        VerticalAlignment="Center"
        Orientation="Vertical"
        Spacing="10">

                
        <StackPanel Orientation="Horizontal" Spacing="5">
          <Button Width="150" Height="30" x:Name="Button_Source" Click="Button_Source_Click">Source Folder:</Button>

          
          <TextBox x:Name="TextBox_Source" PlaceholderText="Enter text here" Style="{StaticResource UniformTextBoxStyle}"/>
            
        </StackPanel>

          
        
        <StackPanel Orientation="Horizontal" Spacing="5">
          <Button Width="150" Height="30" x:Name="Button_Destination" Click="Button_Destination_Click">Target Folder:</Button>
          <TextBox Width="500" Height="Auto" x:Name="TextBox_Destination" PlaceholderText="Enter text here" Margin="10,0,0,0"/>
        </StackPanel>

        
        <StackPanel Orientation="Horizontal" Spacing="5">
          <Button Width="150" Height="30" x:Name="Button_Game_Name" Click="Button_Game_Name_Click">Game Name:</Button>
          <TextBox Width="500" Height="Auto" x:Name="TextBox_Game_Name" PlaceholderText="galaga88" Margin="10,0,0,0"/>
        </StackPanel>

        
        <StackPanel Orientation="Horizontal" Spacing="5">
          <TextBox Width="500" Height="Auto" x:Name="TextBox_Output" PlaceholderText="" Margin="165,0,0,0"/>
         </StackPanel>

        
        <Grid Width="660">
          <Grid.ColumnDefinitions>
            <ColumnDefinition Width="*" />
            <ColumnDefinition Width="*" />
            <ColumnDefinition Width="*" />
          </Grid.ColumnDefinitions>
          <Button Width="150" Height="30" x:Name="Button_Set_Folders" Click="Button_Set_Folders_Click" Grid.Column="0" HorizontalAlignment="Left">Set Folders</Button>
          <Button Width="150" Height="30" x:Name="Button_Copy_Files" Click="Button_Copy_Files_Click" Grid.Column="1" HorizontalAlignment="Center">Copy Files</Button>
          <Button Width="150" Height="30" x:Name="Button_Exit" Click="Button_Exit_Click" Grid.Column="2" HorizontalAlignment="Right">Exit</Button>
        </Grid>

      </StackPanel>
    </Grid>
  </ScrollViewer>
</Page>

[MainPage_xaml.txt](https://github.com/user-attachments/files/17903828/MainPage_xaml.txt)

`

**Comment by Xiaoy312** [comment, marked as answer]
This may seems counter-intuitive, but you actually need to specify the Style.BasedOn when attempt to override it. This is not only the case for Uno, but for WinAppSdk as well:
![image](https://github.com/user-attachments/assets/391fcc10-871c-41f6-8818-6597991602cb)

This should fix the problem you are encounting:
```xml
<Style TargetType="TextBox" x:Key="UniformTextBoxStyle" BasedOn="{StaticResource MaterialOutlinedTextBoxStyle}">
```

You can find the `x:Key` of the default style for the Material theme here:
https://platform.uno/docs/articles/external/uno.themes/doc/lightweight-styling.html#resource-keys

**Reply by ward-horsfall** [reply]
Hi [Xiaoy312](https://github.com/Xiaoy312),

Thankyou great work :)

Thanks again,

Ward
